src/render/program.js

```diff
@@ -162,6 +164,8 @@ class Program<Us: UniformBindings> {
         const gl = context.gl;
 
         if (this.failedToCreate) return;
+        // Following line is billing related code. Do not change. See LICENSE.txt
+        if (!isMapAuthenticated(gl)) return;
```

Comment:
Could this be done at a higher level than in the program? I remember you checked the overhead of this call, is it fairly negligible?